src/tdastro/sources/physical_model.py

```diff
@@ -116,6 +116,49 @@ def set_graph_positions(self, seen_nodes=None):
         for effect in self.effects:
             effect.set_graph_positions(seen_nodes=seen_nodes)
 
+    def _redshift_pre_effect(self, observer_frame_times, observer_frame_wavelengths, params):
+        """Calculate the rest frame times and wavelengths needed to give user the observer frame times
+        and wavelengths (given the redshift).
+
+        Parameters
+        ----------
+        observer_frame_times : numpy.ndarray
+            The times at which the observation is made.
+        observer_frame_wavelengths : numpy.ndarray
+            The wavelengths at which the observation is made (in Angstroms).
+        params : `dict`
+            A dictionary of the model's parameters.
+
+        Returns
+        -------
+        tuple of (numpy.ndarray, numpy.ndarray)
+            The rest frame times and wavelengths needed to generate the rest frame flux densities,
+            which will later be redshifted  back to observer frame flux densities at the observer frame
+            times and wavelengths.
+        """
+        observed_times_rel_to_t0 = observer_frame_times - params["t0"]
+        rest_frame_times_rel_to_t0 = observed_times_rel_to_t0 / (1 + params["redshift"])
+        rest_frame_times = rest_frame_times_rel_to_t0 + params["t0"]
+        rest_frame_wavelengths = observer_frame_wavelengths / (1 + params["redshift"])
+        return (rest_frame_times, rest_frame_wavelengths)
+
+    def _redshift_post_effect(self, flux_density, params):
```

Comment:
As above, I think it is more understandable if we just pass in `redshift` directly instead of the dictionary. We can let earlier functions unpack the dictionary as needed.